go/internal/tinder/driver_rdv.go

```diff
@@ -171,6 +171,107 @@ func (c *rendezvousDiscovery) FindPeers(ctx context.Context, ns string, opts ...
 	return chPeer, err
 }
 
+func (c *rendezvousDiscovery) FindPeersAsync(ctx context.Context, chPeer chan<- peer.AddrInfo, ns string, opts ...discovery.Option) error {
+	// Get options
+	var options discovery.Options
+	err := options.Apply(opts...)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		const maxLimit = 1000
+		limit := options.Limit
+		if limit == 0 || limit > maxLimit {
+			limit = maxLimit
+		}
+
+		// Get cached peers
+		var cache *rpCache
+
+		c.peerCacheMux.RLock()
+		cache, ok := c.peerCache[ns]
+		c.peerCacheMux.RUnlock()
+		if !ok {
+			c.peerCacheMux.Lock()
+			cache, ok = c.peerCache[ns]
+			if !ok {
+				cache = &rpCache{recs: make(map[peer.ID]*rpRecord)}
+				c.peerCache[ns] = cache
+			}
+			c.peerCacheMux.Unlock()
+		}
+
+		cache.mux.Lock()
+		defer cache.mux.Unlock()
+
+		// Remove all expired entries from cache
+		currentTime := time.Now().Unix()
+		newCacheSize := len(cache.recs)
+
+		for p := range cache.recs {
+			rec := cache.recs[p]
+			if rec.expire < currentTime {
+				newCacheSize--
+				delete(cache.recs, p)
+			}
+		}
+
+		cookie := cache.cookie
+
+		// Discover new records if we don't have enough
+		if newCacheSize < limit {
+			// TODO: Should we return error even if we have valid cached results?
```

Comment:
It's not by me btw, it just got caught in the copy paste :
https://github.com/berty/berty/blob/b489e357b902b0a1b3bb2449b45ffa2768fd8ec6/go/internal/tinder/driver_rdv.go#L124
I know that redundant code but there is no clean way to merge them without introducing extra of unintuitive branching (it could have a `findPeers` method with an `sync bool` params, `FindPeers` would run it syncronously with `sync == true` and reverse for `FindPeersAsync`, and the sync params would be used to trigger or not some things (goroutine or regular call of the lambada, closing of the chans or not and silent death vs loud one, ...)